### PR TITLE
Updated to support Postgres.

### PR DESCRIPTION
**Pull request by drdozer**
Added initial support for Postgres. See #3.

### CODE REVIEW

File: src/main/scala/de/mukis/sbt/slick/Keys.scala [hunk 1]
```diff
@@ -19,12 +20,13 @@ trait Keys {
 
   val slickDriver = SettingKey[String]("slick-driver", "The slick driver, e.g. 'scala.slick.driver.MySQLDriver'")
   val slickJDBCDriver = SettingKey[String]("slick-jdbc-driver", "The jdbc driver, e.g. 'com.mysql.jdbc.Driver'")
-  val slickUrl = SettingKey[MakeUrl]("slick-url", "URL to database, e.g. 'jdbc:mysql://localhost:3306/your-db'")
+  val slickUrl = TaskKey[MakeUrl]("slick-url", "URL to database, e.g. 'jdbc:mysql://localhost:3306/your-db'")
+  val slickArguments = TaskKey[MakeArguments]("slick-arguments", "Arguments array for database")
   val slickUser = SettingKey[Option[String]]("slick-user", "User to access database")
   val slickPassword = SettingKey[Option[String]]("slick-password", "Password to access database")
-  val slickPort = SettingKey[Int]("slick-port", "Port to access database")
```
Comment (muuki88) [inline src/main/scala/de/mukis/sbt/slick/Keys.scala:25]: Why change this to as task? `slickPort` doesn't depend on any task?

File: src/main/scala/de/mukis/sbt/slick/Keys.scala [hunk 2]
```diff
@@ -19,12 +20,13 @@ trait Keys {
 
   val slickDriver = SettingKey[String]("slick-driver", "The slick driver, e.g. 'scala.slick.driver.MySQLDriver'")
   val slickJDBCDriver = SettingKey[String]("slick-jdbc-driver", "The jdbc driver, e.g. 'com.mysql.jdbc.Driver'")
-  val slickUrl = SettingKey[MakeUrl]("slick-url", "URL to database, e.g. 'jdbc:mysql://localhost:3306/your-db'")
+  val slickUrl = TaskKey[MakeUrl]("slick-url", "URL to database, e.g. 'jdbc:mysql://localhost:3306/your-db'")
+  val slickArguments = TaskKey[MakeArguments]("slick-arguments", "Arguments array for database")
```
Comment (muuki88) [inline src/main/scala/de/mukis/sbt/slick/Keys.scala]: By introducing the `slickArguments` every user can break the call order of the code-gen main.

I would recommend to use the `slickUser` and `slickPassword` and append in in the `SlickCodeGenPlugin` to the arguments.

File: src/main/scala/de/mukis/sbt/slick/SlickCodeGenH2.scala [hunk 3]
```diff
@@ -15,6 +16,15 @@ object SlickCodeGenH2 extends AutoPlugin {
...
       s"jdbc:h2:mem${db.map(":" + _) getOrElse ""}"
+    },
+    slickArguments := { database =>
+      val driver = slickDriver.value
+      val jdbc = slickJDBCDriver.value
+      val url = slickUrl.value
+      val outputDir = (sourceManaged in Compile).value.getPath
+      val rootPackage = slickPackage.value
+
+      Array(driver, jdbc, url(database), outputDir, rootPackage)
```
Comment (muuki88) [inline src/main/scala/de/mukis/sbt/slick/SlickCodeGenH2.scala]: This order has to be correct in all other plugins as well, which is error prone if changes need to be made.

**Comment by drdozer** [comment]
The issue is that postgres requires credentials as extra arguments but mysql encodes them in the URL. How about this? Make the url key into Option[String]. Then have a key that holds a binary flag - postgrestyle/mysqlstyle. Use the binary flag to switch the url-generating and arguments behaviour. That way a person can't do it too badly wrong manually.

**Comment by drdozer** [comment]
If the port is dynamically reconfigured, then it has to be a TaskKey. For example, I'm dynamically assigning the port because it is provided by a randomy bound port from a docker container.

**Comment by drdozer** [comment]
The postgres plugin works now. I'm going to refactor to get rid of the slickArguments key as you suggest.

**Comment by drdozer** [comment]
I'm considering re-writing the URL bit of the plugin to either take a url string, or if this is None, generate it from keys. So slickUrl becomes Option[String], and if it is None, a plugin-specific function kicks in to generate a URL using all the keys.

**Comment by drdozer** [comment]
OK, the current head works and seems OK to me. No need at this time to re-visit the URL issue I think.

**Comment by muuki88** [comment]
LGTM

**Comment by muuki88** [comment]
released with `0.5`

**Pull request merged**